r/R/type.R

```diff
@@ -423,6 +423,24 @@ as_type <- function(type, name = "type") {
   type
 }
 
+canonical_type_str <- function(type_str) {
+  # canonicalizes data type strings, converting aliases to match the strings
```

Comment:
In some senses I think this is backwards: IIUC "float64" is canonical, but (for some reason) the C++ library prints it as "double".

Reply:
Per https://arrow.apache.org/docs/cpp/api/datatype.html it looks like "double" is canonical

Reply:
🤷 Alrighty then, proceed